apps/vscode/test_server.py

```
@@ -295,12 +347,168 @@ def run_tests(base_dir: Path) -> bool:
+        script_path = Path(script_file.name)
+
+    env = os.environ.copy()
+    env.setdefault("ADMIN_USERNAME", "admin")
+    env.setdefault("ADMIN_PASSWORD", os.environ.get("DOCS_ADMIN_PASSWORD", "admin"))
+    env.setdefault(
+        "ADMIN_URL", f"http://localhost:{SCREENSHOT_PORT}/admin/login/?next=/admin/"
+    )
+    env.setdefault("PUBLIC_URL", f"http://localhost:{SCREENSHOT_PORT}/")
+
+    try:
+        result = subprocess.run(["node", str(script_path)], cwd=base_dir, env=env)
```

Ruff (0.15.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)